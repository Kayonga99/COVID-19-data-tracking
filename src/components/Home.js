/* eslint-disable */

import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchData } from '../redux/covidData/covid';

function Home() {
  const covidReducer = useSelector((state) => state.covidReducer.covid);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!covidReducer.length) {
      dispatch(fetchData());
    }
  }, []);
  const Europe = covidReducer.filter((item) => item.continent === 'Europe')
  return (
    <div className='body'>
      
        {
        Europe.map((data) => {
          const {
            country,
            flag,
          } = data;

          return (
            <Link key={country} to={{pathname: `/country/${country}`}}>


            <div className="main-cont" key={country}>
              {country}
              <img src={flag} alt={country} />

            </div>
            </Link>

          );
        })
      }
    </div>

  );
}

export default Home;
