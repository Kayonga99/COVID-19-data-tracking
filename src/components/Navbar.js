import React from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import { TiArrowBackOutline } from 'react-icons/ti';
import mic from '../images/mic-svgrepo-com.svg';
import set from '../images/settings-svgrepo-com.svg';

function Navbar() {
  const place = useLocation();
  const back = <TiArrowBackOutline className="back" />;
  const goBack = place.pathname.includes('country') ? back : '';
  return (
    <nav className="navbar">
      <div>
        <NavLink exact="true" to={{ pathname: '/' }}>
          {goBack}
        </NavLink>
      </div>
      <h2>COVID-19 DATA</h2>
      <div className="top-right">

        <div className="top-right">
          <img src={mic} className="mic" alt="" />
        </div>
        <div className="top-right">
          <img className="set" src={set} alt="#" />
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
