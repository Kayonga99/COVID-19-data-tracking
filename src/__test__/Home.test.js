import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';
import Home from '../components/Home';

describe('Render components', () => {
  test('render Navbar', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Navbar /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render Home', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Home /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});