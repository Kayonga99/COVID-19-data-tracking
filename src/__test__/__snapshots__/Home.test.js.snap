// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render Home 1`] = `
Array [
  <div
    className="heading"
  >
     
    <h1>
      EUROPE COUNTRIES' COVID-19 DATA
    </h1>
     
  </div>,
  <div
    className="input-box"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Serch European Country ..."
      type="text"
      value=""
    />
  </div>,
  <div
    className="body"
  />,
]
`;

exports[`Render components render Navbar 1`] = `
<nav
  className="navbar"
>
  <div>
    <a
      aria-current="page"
      className="active"
      exact="true"
      href="/"
      onClick={[Function]}
    />
  </div>
  <h2
    className="c-data"
  >
    COVID-19 DATA
  </h2>
  <div
    className="top-right"
  >
    <div
      className="top-right"
    >
      <img
        alt=""
        className="mic"
        src="mic-svgrepo-com.svg"
      />
    </div>
    <div
      className="top-right"
    >
      <img
        alt="#"
        className="set"
        src="settings-svgrepo-com.svg"
      />
    </div>
  </div>
</nav>
`;
